src/main/webapp/feed.html

```diff
@@ -23,24 +23,48 @@ <h1>Message Feed</h1>
 
     function MessageFeed() {
       const [allMessages, setMessages] = React.useState([])
+      const [numMessagesPerUser, setNumUserMessages] = React.useState(["No messages!", h('br')])
```

Comment:
I think this solution is suboptimal since here you have to keep track of two sources of truth when it comes to state -- a list of React elements which contain users messages, and also an object which keeps track of how many messages each user sent. Can you find a way to reconcile these two so you only have to call `useState` once?

Reply:
If you do do this, it should turn this ~50 line component into a ~15 line component.